### ISSUE TITLE
Crash of the 1.6.0 on WatchOS 8

### ISSUE THREAD

**Issue by djavan-bertrand** [amplitude/Amplitude-Swift#178]
The SDK 1.6.0 crashes on watchOS 8. 
The crash occurs at app startup:

```
dyld[87977]: Symbol not found: _$sSo13WKApplicationC8WatchKitE30didEnterBackgroundNotificationSo18NSNotificationNameavgZ
  Referenced from: /Users/djavan.bertrand/Library/Developer/CoreSimulator/Devices/E858B752-0788-40E6-9BD4-F16B5FF6579B/data/Containers/Bundle/Application/7BE69845-5B5D-4047-A6EB-18D887720C21/Watch-App-Sampler WatchKit App.app/PlugIns/Watch-App-Sampler WatchKit Extension.appex/Watch-App-Sampler WatchKit Extension
  Expected in: /Library/Developer/CoreSimulator/Profiles/Runtimes/watchOS 8.5.simruntime/Contents/Resources/RuntimeRoot/usr/lib/swift/libswiftWatchKit.dylib
```

Please note that it does not crash with the 1.5.2.

## Possible Solution
I don't have a solution, but this might be related to the changes brought in this PR: https://github.com/amplitude/Amplitude-Swift/pull/163
Maybe using compiler directive or availability directives (something like `if #available(watchOS 8.0, *)` would solve this crash?

## Steps to Reproduce
Please use the following sample project (it is just a watch OS app with a dependency on Amplitude-Swift 1.6.0):
[Amplitude-Watch-Test.zip](https://github.com/user-attachments/files/15774218/Amplitude-Watch-Test.zip)

1. Run it on a watchOS 8 simulator or real device

## Environment
- SDK Version: 1.6.0
- Xcode: Xcode 15.4

**Comment by thedavidharris** [comment]
Oof, apparently this is... a mess. https://github.com/RevenueCat/purchases-ios/pull/1895 I think has an example for how to fix this.

Will likely not get a chance tomorrow but can probably PR this before the end of the week. Sorry for the confusion here.

**Comment by crleona** [comment]
HI @djavan-bertrand, @thedavidharris - I have a [patch](https://github.com/amplitude/Amplitude-Swift/pull/180) which should resolve this issue and another issue tracking sessions on watchOS.

**Comment by djavan-bertrand** [comment]
@crleona Hi, thanks for this quick work. I've tried the patch on devices (simulator and real devices) and it seems to work. 💪

**Comment by crleona** [comment]
This should be released in 1.6.1.  @thedavidharris @djavan-bertrand thank you for your help!
